R/geom-rug.r

```diff
@@ -76,7 +80,10 @@ geom_rug <- function(mapping = NULL, data = NULL,
 GeomRug <- ggproto("GeomRug", Geom,
   optional_aes = c("x", "y"),
 
-  draw_panel = function(data, panel_params, coord, sides = "bl", outside) {
+  draw_panel = function(data, panel_params, coord, sides = "bl", length, outside) {
```

Comment:
It's generally better to add new arguments (here, `length`) after previously existing arguments, in case some downstream code depends on the order of the arguments. Also, better to provide a default length here (and maybe also add a default `outside` value).

Reply:
Should I change the ordering in the `geom_rug` function as well as, or instead of, `draw_panel`? It seems changing the order or defaults for draw_panel makes no difference when adding geom_rug() to a plot.

Reply:
No. In `geom_rug()`, `length` is placed after `...`, which means the user need to pass the argument with its name. So, there's no chance for `length` to be mistaken. But, in `draw_panel()`, `outside` can be specified without the name. If some existing code uses `outside` positionally, it might fail due to this change.

Does this answer your question?

(Sorry that I didn't notice this on my review...)

Reply:
Ah I see thanks! I've implemented the three new suggestions.